src/main/java/org/concordion/internal/FixtureRunner.java

```diff
@@ -12,8 +13,8 @@
     private static RunResultsCache runResultsCache = RunResultsCache.SINGLETON;
     private Concordion concordion;
 
-    public FixtureRunner(Fixture fixture) throws UnableToBuildConcordionException {
-        ConcordionBuilder concordionBuilder = new ConcordionBuilder().withFixture(fixture);
+    public FixtureRunner(Fixture fixture, SpecificationLocator specificationLocator) throws UnableToBuildConcordionException {
```

Comment:
Rather than changing the signature of this method, and introducing a back-way of adding the specificationLocator, could the `JpaUnitConcordionRunner` use reflection to add the specificationLocator as an `@Extensions` annotation to the fixture? This would then be loaded through the standard `FixtureExtensionLoader()`, making most of these changes unnecessary.

Reply:
It could use reflection to achieve this, but it would be by the cost of introducing Java version and potentially JVM implementation dependency. E.g. `java.lang.Class$AnnotationData`, which defines the type used to store Annotation related data and the field, respectively method `java.lang.Class.annotationsData` were e.g. introduced with Java 8. In Java 7 the name of this field was just `java.lang.Class.annotations`. The access to this field would be required to insert an annotation at run-time. I would like to avoid dependencies like described above.

Reply:
Fair enough. It looks like it can be achieved with Java Proxies such as in the [Javanna](https://github.com/renatoathaydes/javanna/blob/master/src/main/java/com/athaydes/javanna/Javanna.java) library, but not easy.

Before I merge your PR, I just wanted to check on another option - adding the fully qualified name of an extension containing the specificationLocator to the System property `concordion.extensions`. I realise this would change it for all Concordion tests in the current JVM, so may be more broad-reaching than you need?

Let me know, and I'll merge your PR if that's the best option. You will need to agree to the CLA however.

Reply:
Javanna could be used to to create the actual annotation instance, but will not solve the root issue of accessing Java version and potentially JVM implementation specific fields. It might also become more funny with Java 9.

Give me please some time to check your idea of using the `concordion.extension` property.